### ISSUE TITLE
Array in plugin config fails instanceof Array check

### ISSUE THREAD

**Issue by unional** [karma-runner/karma#1578]
I'm using `karma-systemjs` and found a bug that most likely an issue on `karma` side, so want to bring it up to your attention.

In `karma-systemjs`, it will configure `systemjs` using a config file. In the config, there is a section with an array:
```
meta: {
  "*": {
    deps: ['jquery', 'moduleB', ...]
  }
}
```
`karma-systemjs` loads the config and pass it to the browser. Before karma, `instanceof Array` is working fine:

// code simplified
`console.log(config.meta['*'].deps instanceof Array) => true`

But inside karma, it fails:
```
console.log(karma.config.systemjs.config.meta['*'].deps, karma.config.systemjs.config.meta['*'].deps instanceof Array);
```
produces:
```
LOG: ['jquery', 'moduleB', ...], false
```
https://github.com/rolaveric/karma-systemjs/issues/44

**Comment by Dignifiedquire** [comment]
Thanks for the report, it looks like this should be passed through as an Array, I agree. As far as I know this serialization is handled by socket.io, which should be using `JSON.stringify/JSON.parse`, but maybe there is some point where things do are not passed as they should be.

**Comment by unional** [comment]
Hi @Dignifiedquire ,

@nkzawa mentioned `instanceof Array` does not work across iframe:
https://github.com/socketio/socket.io/issues/2234#issuecomment-172320250

http://stackoverflow.com/questions/22289727/difference-between-using-array-isarray-and-instanceof-array

My guess is it is not the case here, as those code are in `karam-systemjs` and not access through any tests (does tests run in iframe?).

Can you confirm?

**Comment by Dignifiedquire** [comment]
Yes the tests are run inside an iframe, so this might very well be the referenced stackoverflow issue. Did you test with `Array.isArray` yet?

**Comment by unional** [comment]
Just to confirm, while tests are run in iframe, do plugins also run in iframe?

The code is run before calling `karma.start()`,
https://github.com/rolaveric/karma-systemjs/blob/master/lib/adapter.js#L80-L104

**Comment by Dignifiedquire** [comment]
That part is run outside the iframe, but the tests are executed inside the iframe

**Comment by unional** [comment]
Then it should not be iframe issue.

**Comment by unional** [comment]
per https://github.com/karma-runner/karma/issues/1578#issuecomment-172803831

**Comment by unional** [comment]
@Dignifiedquire , how does `karma` beam the plugin config up to the client? Is it something similar to what @nkzawa describe?
https://github.com/socketio/socket.io/issues/2234#issuecomment-172320211

I'm not familiar with `karma` code, can out point out where that code is located?

Thanks,
